src/login/LoginViewController.js

```diff
@@ -154,56 +154,30 @@ export class LoginViewController implements ILoginViewController {
 			            this.view.helpText = lang.get('emptyString_msg')
 			            m.redraw()
 		            })
-		            .catch(AccessBlockedError, e => {
-			            this.view.helpText = lang.get('loginFailedOften_msg')
+		            .catch(e => {
+			            this.view.helpText = lang.get(getLoginErrorMessage(e))
 			            m.redraw()
-			            return errorAction()
-		            })
-		            .catch(BadRequestError, e => {
-			            this.view.helpText = lang.get('loginFailed_msg')
-			            this.view.invalidCredentials = true
-			            m.redraw()
-			            return errorAction()
-		            })
-		            .catch(NotAuthenticatedError, e => {
-			            this.view.helpText = lang.get('loginFailed_msg')
-			            this.view.invalidCredentials = true
-			            m.redraw()
-			            return errorAction()
-		            })
-		            .catch(AccessDeactivatedError, e => {
-			            this.view.helpText = lang.get('loginFailed_msg')
-			            m.redraw()
-			            return errorAction()
-		            })
-		            .catch(AccessExpiredError, e => {
-			            this.view.helpText = lang.get('inactiveAccount_msg')
-			            this.view.accessExpired = true
-			            m.redraw()
-			            return errorAction()
-		            })
-		            .catch(TooManyRequestsError, e => {
-			            this.view.helpText = lang.get('tooManyAttempts_msg')
-			            m.redraw()
-			            return errorAction()
-		            })
-		            .catch(CancelledError, () => {
-			            this.view.helpText = lang.get('emptyString_msg')
-			            m.redraw()
-			            return errorAction()
-		            })
-		            .catch(ConnectionError, e => {
-			            if (client.isIE()) {
-				            // IE says it's error code 0 fore some reason
-				            this.view.helpText = lang.get('loginFailed_msg')
-				            m.redraw()
+
+			            this.view.invalidCredentials =
+				            this.view.invalidCredentials || e instanceof BadRequestError || e instanceof NotAuthenticatedError
```

Comment:
Why is it OR with previous value? Can it go `false` then?

Reply:
I see, maybe just "if" and then not set it otherwise? it will be more like before.

Reply:
it's the same, i just thought it was nice this way. if we can prefer i can change to ifs